crates/ide/src/navigation_target.rs

```diff
@@ -165,7 +199,15 @@ impl TryToNav for FileSymbol {
 
         Some(NavigationTarget {
             file_id: full_range.file_id,
-            name: self.name.clone(),
+            name: if self.is_alias {
+                NavigationTargetName::Alias(
+                    self.name.clone(),
+                    // FIXME : Can we guarantee that canonical path always exists for FileSymbols?
+                    SmolStr::from(self.def.canonical_path(db)?),
```

Comment:
Given my comment on adding an alias field, to get the original name we can just do `self.def.name(db)` here then. That is still an optional value, but the only 2 cases where there is no name is either for an unnamed constant (which has name `_`) or for the crate root module which has no name. Those two should never make it to this point anyways so that's fine to return via `?` then